azurelinuxagent/ga/exthandlers.py

```diff
@@ -1399,6 +1439,43 @@ def get_seq_no(self):
         # Will have to revisit once the feature to enable multiple runtime settings is rolled out by CRP
         return self.ext_handler.properties.extensions[0].sequenceNumber
 
+    def _read_and_parse_json_status_file(self, ext, seq_no, ext_status_file):
+        failed_to_read = False
+        failed_to_parse_json = False
+        raised_exception = None
+        data_str = None
+        data = None
+
+        for attempt in range(_NUM_OF_STATUS_FILE_RETRIES):
+            try:
+                data_str = fileutil.read_file(ext_status_file)
+                data = json.loads(data_str)
+                break
+            except IOError as e:
+                failed_to_read = True
+                raised_exception = e
+            except ValueError as e:
+                failed_to_parse_json = True
+                raised_exception = e
+            time.sleep(_STATUS_FILE_RETRY_DELAY)
+
+        if failed_to_read:
+            msg = u"We couldn't find any status for {0}-{1} extension, for the sequence number {2}. It failed due to " \
+                  u"{3}".format(ext.name, self.ext_handler.properties.version, seq_no, raised_exception)
+            raise ExtensionStatusError(msg=msg, code=ExtensionStatusError.StatusFileNotFound)
+        elif failed_to_parse_json:
+            msg = u"Failed to read any status for extension - {0}-{1}, Sequence number {2}. Failed due to {3}" \
```

Comment:
similarly, this would be "Failed to parse", and also try to make it more user-friendly. Maybe something like "Extension foo reported its status using an incorrect format", or something along those lines.

Reply:
Done.